### PR TITLE
Closes #505

### LINKED ISSUE #505: Proposal to add MedQA
## Adding a Dataset
- **Name:** MedQA
- **Description:** "In this work, we present the first free-form multiple-choice OpenQA dataset for solving medical problems, MedQA,
collected from the professional medical board exams. It covers three languages: English, simplified Chinese, and
traditional Chinese, and contains 12,723, 34,251, and 14,123 questions for the three languages, respectively. Together
with the question data, we also collect and release a large-scale corpus from medical textbooks from which the reading
comprehension models can obtain necessary knowledge for answering the questions."
- **Task:** QnA
- **Paper:** https://www.mdpi.com/2076-3417/11/14/6421
- **Data:** https://github.com/jind11/MedQA
- **License:** ?
- **Motivation:** Large scale OpenQA dataset in three different languages, also providing textbooks

### PR DESCRIPTION
**Pull request by nomisto**
Closes #505

Dataset would also provide two textbooks (One english, One Chinese). Can I include them somehow?

### Checkbox

- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `biodatasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_BIGBIO_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `BigBioConfig` for the source schema and one for a bigbio schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_bigbio biodatasets/my_dataset/my_dataset.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: biodatasets/med_qa/med_qa.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+                example = example.to_dict()
+                example["options"] = [{"key": key, "value": value} for key, value in example["options"].items()]
+                yield key, example
+
+        elif self.config.schema == "bigbio_qa":
+            for key, example in data.iterrows():
+                example = example.to_dict()
+                example_ = {}
+                example_["id"] = str(key)
+                example_["question_id"] = ""
```
Comment (ruisi-su) [inline biodatasets/med_qa/med_qa.py:206]: Could you please use the same value as "id" for the `question_id` and `document_id`? I quickly skimmed through merged QA [dataloaders](https://github.com/bigscience-workshop/biomedical/blob/master/biodatasets/pubmed_qa/pubmed_qa.py#L247), and this seems to be the way that folks dealt with this. Thank you for working on this!

**Review by ruisi-su — CHANGES_REQUESTED** [review]

File: biodatasets/med_qa/med_qa.py (hunk 1 above)
Comment (nomisto) [reply biodatasets/med_qa/med_qa.py]: Sure, I've changed it!

**Comment by nomisto** [comment]
@ruisi-su The dataset includes two moderately sized textbooks (~250 mb) that can/should be used as corpora for these questions. Do you think I can include them as a source config, f.e. `med_qa_textbook_english_source` or `med_qa_textbook_chinese_source` without a bigbio schema? Or should I omit the corpora completely?

**Comment by ruisi-su** [comment]
@jason-fries @hakunanatasha any thoughts on the @nomisto's question about adding corpora to the dataset?

**Review by ruisi-su — APPROVED** [review]
LGTM, unit tests also pass with this command:
` python -m tests.test_bigbio_by_name biodatasets/med_qa/med_qa.py med_qa_zh_source --data_dir <local_dir>` Thanks again @nomisto for your contribution!

**Comment by hakunanatasha** [comment]
@ruisi-su if this passes the tests, please feel free to merge.

**Pull request merged**